private/buf/bufformat/testdata/proto3/all/v1/all.golden.proto

```diff
@@ -0,0 +1,117 @@
+// top-level comment 1
+// top-level comment 2
+
+syntax = "proto3"; // syntax-level-inline comment
+
+// between-syntax-and-package comment
+package all.v1; // package-level-inline comment
+
+import "custom.proto";
+// between-imports comment
+import "google/protobuf/duration.proto"; // import-inline comment  2
+// between-package-and-import comment
+// import comment
+import "google/protobuf/timestamp.proto"; // import-inline comment
+
+// option comment 5
+option (custom.file_option) = true; // option-inline comment 5
+// option comment 6
+option (custom.file_thing_option) = {foo:1 bar:2 truth:false recursive:{foo:3 bar:4 truth:true}}; // option-inlinecomment 6
+// option comment 3
+option go_package = "foopb"; // option-inline comment 3
+// between-imports-and-options comment
+// option comment
+option java_multiple_files = true; // option-inline comment
+// option comment 4
+option java_outer_classname = "FooProto"; // option-inline comment 4
+// option comment 2
+option java_package = "com.foo"; // option-inline comment 2
+
+// leading comment one1
+// leading comment one2
+message One {
+  // message-option comment
+  // subsequent message-option comment
+  option (custom.message_option) = false; // message-option-inline comment
+  // message-option comment 2
+  option (custom.message_thing_option) = {foo:5}; // message-option-inline comment 2
+
+  // body comment 2
+  // body comment 3
+  // nested-message comment
+  message Three {
+    int64 three_one = 1;
+  }
+
+  /*
+  c-style comment
+  */
+  int64 one_one = 1 [(custom.field_option) = true];
```

Comment:
@amckinney for alignment we found the following [clang-format rules](https://clang.llvm.org/docs/ClangFormatStyleOptions.html) quite convenient:

```
AlignConsecutiveDeclarations: true
AlignConsecutiveAssignments: true
```

For this few lines formatting would look like this:

```proto
  int64              one_one   = 1 [(custom.field_option) = true];
  Two                one_two   = 2 [(custom.field_option) = false, (custom.field_thing_option) = {foo:6}]; // field-option-inline comment
  Three              one_three = 3; /* field-inline c-style comment */
  map<string,uint32> map_field = 4;
```

<details><summary>Click to show longer example</summary>
<p>

#### From one of our public proto

```proto
message CreateClusterRequest {
  message OpenIDConnectConfig {
    string                      issuer_url      = 1;
    string                      client_id       = 2 [(validate.rules).string.min_len = 1];
    google.protobuf.StringValue username_claim  = 3 [(validate.rules).string.min_len = 1];
    google.protobuf.StringValue username_prefix = 4 [(validate.rules).string.min_len = 1];
    scaleway.std.StringsValue   groups_claim    = 5;
    google.protobuf.StringValue groups_prefix   = 6 [(validate.rules).string.min_len = 1];
    scaleway.std.StringsValue   required_claim  = 7;
  }

  message PoolConfig {
    message UpgradePolicy {
      google.protobuf.UInt32Value max_unavailable = 1;
      google.protobuf.UInt32Value max_surge       = 2;
    }

    string                      name               = 1 [(validate.rules).string.min_len = 1];
    string                      node_type          = 2;
    google.protobuf.StringValue placement_group_id = 3 [(validate.rules).string.uuid = true];
    bool                        autoscaling        = 4;
    uint32                      size               = 5;
    google.protobuf.UInt32Value min_size           = 6;
    google.protobuf.UInt32Value max_size           = 7;
    Runtime                     container_runtime  = 8 [(validate.rules).enum.defined_only = true];
    bool                        autohealing        = 9;
    repeated string             tags               = 10;
    map<string, string>         kubelet_args       = 11;
    UpgradePolicy               upgrade_policy     = 12;
    string                      zone               = 13;
  }

  message AutoscalerConfig {
    google.protobuf.BoolValue   scale_down_disabled              = 1;                                             // default to false
    google.protobuf.StringValue scale_down_delay_after_add       = 2;                                             // default to 10m
    AutoscalerEstimator         estimator                        = 3 [(validate.rules).enum.defined_only = true]; // default to binpacking
    AutoscalerExpander          expander                         = 4 [(validate.rules).enum.defined_only = true]; // default to random
    google.protobuf.BoolValue   ignore_daemonsets_utilization    = 5;                                             // default to false
    google.protobuf.BoolValue   balance_similar_node_groups      = 6;                                             // default to false
    google.protobuf.Int32Value  expendable_pods_priority_cutoff  = 7;                                             // default to -10
    google.protobuf.StringValue scale_down_unneeded_time         = 8;                                             // default to 10m
    google.protobuf.FloatValue  scale_down_utilization_threshold = 9 [(validate.rules).float = { gt: 0, lt: 1 }]; // default to 0.5
    google.protobuf.UInt32Value max_graceful_termination_sec     = 10;                                            // default to 600
  }

  message AutoUpgrade {
    bool              enable             = 1;
    MaintenanceWindow maintenance_window = 2;
  }

  oneof ProjectIdentifier {
    string organization_id = 1 [(validate.rules).string.uuid = true, deprecated = true];
    string project_id      = 14 [(validate.rules).string.uuid = true];
  }

  string              type                   = 17;
  string              name                   = 2 [(validate.rules).string.min_len = 1];
  string              description            = 3;
  string              version                = 4;
  repeated string     tags                   = 5;
  CNI                 cni                    = 6 [(validate.rules).enum.defined_only = true];
  bool                enable_dashboard       = 7 [deprecated = true];
  Ingress             ingress                = 8 [(validate.rules).enum.defined_only = true];
  repeated PoolConfig pools                  = 9;
  AutoscalerConfig    autoscaler_config      = 10;
  AutoUpgrade         auto_upgrade           = 11;
  repeated string     feature_gates          = 12;
  repeated string     admission_plugins      = 13;
  OpenIDConnectConfig open_id_connect_config = 15;
  repeated string     apiserver_cert_sans    = 16;
}

```

</p>
</details>

Do you think it's interesting to add this kind of formatting option in the buf formater?

Reply:
Nice, I think this is totally reasonable. I'd like to start with with a configuration-less formatter implementation, but I imagine that we'll follow up with shortly after with a `format` configuration (see summary above for details).

I'll make note of this for future changes - thanks for the suggestion!

Reply:
Opened a PR to keep track of this https://github.com/bufbuild/buf/issues/1312